### ISSUE TITLE
Ruby/Project: Event Manager formatting, gem deprecation, API key and unused file

### ISSUE THREAD

**Issue by scheals** [TheOdinProject/curriculum#23552]
[The project is adapted from another resource](https://www.theodinproject.com/paths/full-stack-ruby-on-rails/courses/ruby-programming/lessons/event-manager-ruby-programming)
>Please note this tutorial has been adapted from The Turing School’s and Jump Start Lab’s [Event Manager](http://tutorials.jumpstartlab.com/projects/eventmanager.html) and updated to use GoogleCivic API

and this makes these issues appear:
- There seems to be some inconsistency with how it is formatted compared to the rest of the course, `assignment` and `iteration` blocs probably could use some formatting like becoming headers of some sort.
- [It is using project source's API key](https://www.theodinproject.com/paths/full-stack-ruby-on-rails/courses/ruby-programming/lessons/event-manager-ruby-programming#accessing-the-api), probably best to get an Odin one instead if the learner is not supposed to create their own for the project?
> When we’re accessing the representatives method of their API, we’re sending in a key which is the string that identifies JumpstartLab as the accessor of the API,
- In [Initial Setup](https://www.theodinproject.com/paths/full-stack-ruby-on-rails/courses/ruby-programming/lessons/event-manager-ruby-programming#initial-setup) we're given links to two samples but the tutorial only ever uses one. Should the `large sample` just be removed or the project be changed to make use of it?
- The gem installed in the project is deprecated and the suggested one, `google-apis-civicinfo_v2` seems to be working just fine without it but I'd love additional eyes on this since my gem-fu isn't the strongest. I removed all my gems and installed only the aforementioned `google-apis-civicinfo_v` and it worked like a charm without any problems. No issues with `signet` too, this might be connected to changes they've made to the gems. They got updated to Ruby 3 in May 2021 and you now use particular gems for particular APIs. Again, my knowledge is lacking but the link about downgrading goes to a [GitHub issue](https://github.com/googleapis/google-api-ruby-client/issues/833) where I found out about the update and it seems like that this might not occur any more given we're on Ruby 3.0.3.

**Comment by wise-king-sullyman** [comment]
Sounds like a decent amount of work...

@KevinMulhern are there any plans currently involving this lesson in the Ruby reworking you have planned?

**Comment by KevinMulhern** [comment]
Sorry just catching up with GH @wise-king-sullyman, no plans at the moment. Our focus is on the Rails course. But this has been added to our backlog to have a look at.

**Comment by ChargrilledChook** [comment]
Some new but related issues have started popping up with this lesson, perhaps due to some broken dependencies in the gem
https://discord.com/channels/505093832157691914/995029097103249508

**Comment by ManonLef** [comment]
@TheOdinProject/ruby is this still relevant?

**Comment by rlmoser99** [comment]
Yes, I believe this is still needed to do. It is has not been a high priority.

**Comment by SumonGFC** [comment]
Hello, I would like to help resolve this issue. 
- [ ] Fix formatting of `assignment` and `iteration` blocks (and also maybe the markdown lint error `TOP004/lesson-headings Required heading structure [Context: "### Assignment]` ?)

- [x] Use TOP key instead of the project source's key (Correct me if I'm wrong but this seems to be solved since [the key we have in the lesson](https://github.com/TheOdinProject/curriculum/blob/main/ruby/files_and_serialization/project_event_manager.md?plain=1#L611)  is different from [Jumpstart Lab's key](http://tutorials.jumpstartlab.com/projects/eventmanager.html#:~:text=Iteration%203%3A%20Using%20Sunlight) (which doesn't even use google-apis), and [the Turing School lesson](https://backend.turing.edu/module1/projects/eventmanager) doesn't even go into APIs at all.

- [ ] Fix [Initial setup large/small sample data](https://github.com/TheOdinProject/curriculum/blob/main/ruby/files_and_serialization/project_event_manager.md?plain=1#L75-L76) -- The large sample dataset is not mentioned after it is first referenced on line 76. Mention of it should just be removed from the lesson altogether.

- [ ] Update `google-api-client` gem to `google-apis-civicinfo_v2` gem. I am not sure what to do about this one. Will probably require some more testing perhaps? It seems @scheals has already tested this (2 years ago). For what it's worth, I completed this lesson a month or 2 ago and was able to use the deprecated API client without any issues (but it's probably a good idea to update this so that this part of the tutorial doesn't break in the future). At their convenience, if a maintainer (or anyone else) could weigh in on this issue and perhaps offer some guidance, that would be appreciated

If all is well, I can begin to work on the first and third fixes mentioned above (i.e. assignment/iteration formatting, and removing mention of the large sample data dataset).

**Comment by JoshDevHub** [comment]
Hi @SumonGFC apologies for the delay on this.

With #28827 the final point in your checklist has been resolved, so I marked that off for you. I'd be glad to assign you to work on the other two available points in your checklist. Let me know if you still want to work on this.

**Comment by SumonGFC** [comment]
@JoshDevHub Yeah I can get this done. Please assign me.

**Comment by JoshDevHub** [comment]
Thank you for volunteering @SumonGFC . I'll assign you
